src/Controls/src/Core/Registrar.cs

```diff
@@ -54,6 +55,8 @@ public void Register(Type tview, Type trender, Type[] supportedVisuals, short pr
 				}
 				else
 					visualRenderers[supportedVisuals[i]] = (trender, priority);
+
+				RegisterVisual(supportedVisuals[i]);
```

Comment:
You are probably getting ILLink warnings here since the `supportedVisuals[i]` isn't annotated as `PublicParameterlessConstructor`.